APIs_RTOS/Events_ex_1/main.cpp

```diff
@@ -17,30 +19,69 @@ void handler(int count)
 
 void post_events(void)
 {
-
     // Events can be posted multiple times and enqueue gracefully until
-    // the dispatch function is called.
-    event.post(1);
-    event.post(2);
-    event.post(3);
+    // the dispatch function is called. Each event will be processed
+    // subject to any delay and period specified. Each time an event has
+    // been dispatched it will be re-queued ready for the next dispatch
+    // period.
+    event1.post(1);      // Event1 with a value of 1
+    event1.post(2);      // Event1 with a value of 2
+    event1.post(3);      // Event1 with a value of 3
+
+    // Cancel the last event posted ie Event1 with a value of 3
+    event1.cancel();
+
+    event1.post(4);      // Event1 with a value of 4
+
+    event2.post(5);      // Event2 with a value of 5
+
 }
 
+
+// Example  demonstrates the following:
+// 1. Post 5 different events to a queue
+// 2. Configure the event delay and period for each
+// 3. Invoke the dispatcher to dispatch events for a specified period
+//
+//    | 100ms |    200ms    | 100ms | 100ms |    200ms     |
+//            ^             ^       ^       ^              ^
+//       Events 1,2,4       ^       ^       ^              ^
+//        dispatched        ^       ^       ^              ^
+//                          ^       ^       ^              ^
+//                   Events 1,2,4   ^       ^              ^
+//                    dispatched    ^       ^              ^
+//                                  ^       ^              ^
+//                               Event 5    ^              ^
+//                                          ^              ^
+//                                      Event 1,2,4        ^
+//                                       dispatched        ^
+//                                                         ^
+//                                                       Events 1,2,4
+//                                                        dispatched
+//
```

Comment:
That's great. Could you also show the expected output and perhaps print the time in `handler`?

Reply:
Yes that's a good idea

Reply:
I like ascii art to show the sequence. I remember we did it in minar scheduler and it was quite clear what when should be triggered.